src/components/Events/index.js

```diff
@@ -4,25 +4,125 @@
+  const [scrollingKey, setScrollingKey] = useState("");
+
+  /// helpers ///
+  const filterEvents = (eventsToFilter, filterDate) => {
+    return eventsToFilter.filter((event) => event.timestamp > filterDate.getTime()) || [];
+  };
+
   const trimmedEvents = useMemo(() => {
+    let normalizedEvents;
+
     // Event limit is 500, anything more than that will be truncated.
     if (events && events.length && events.length > 500) {
-      return events.slice(0, 500);
+      normalizedEvents = events.slice(0, 500);
     } else {
-      return events;
+      normalizedEvents = events;
     }
-  }, [events]);
+
+    // Filter out events based on timeline filter.
+    if (!!filterValue) {
+      const filterDate = new Date();
+
+      switch (filterValue) {
+        case "Last 15 minutes":
+          filterDate.setMinutes(filterDate.getMinutes() - 15);
+          break;
+        case "Last 30 minutes":
+          filterDate.setMinutes(filterDate.getMinutes() - 30);
+          break;
+        case "Last 1 hour":
+          filterDate.setHours(filterDate.getHours() - 1);
+          break;
+        case "Last 3 hours":
+          filterDate.setHours(filterDate.getHours() - 3);
+          break;
+        case "Last 6 hours":
+          filterDate.setHours(filterDate.getHours() - 6);
+          break;
+        case "Last 12 hours":
+          filterDate.setHours(filterDate.getHours() - 12);
+          break;
+        case "Last 1 day":
+          filterDate.setDate(filterDate.getDate() - 1);
+          break;
+        case "Last 7 days":
+          filterDate.setDate(filterDate.getDate() - 7);
+          break;
+        case "Last 21 days":
+          filterDate.setDate(filterDate.getDate() - 21);
+          break;
+        default:
+          break;
+      }
+
+      const filteredEvents = filterEvents(normalizedEvents, filterDate);
+      if (filteredEvents.length === 0) {
+        // No events for the filter, show an error message and continue to show
+        // all events.
+        setLastFilterHadNoEvents(true);
+      } else {
+        // The filter has events, so we show those events.
+        normalizedEvents = filteredEvents;
+        setLastFilterHadNoEvents(false);
+      }
+    }
+
+    return normalizedEvents;
+  }, [events, filterValue]);
+
+  useEffect(() => {
+    listRefs.current = listRefs.current.slice(0, trimmedEvents.length);
+ }, [trimmedEvents]);
+
+  const plotClickHandler = (eventName, timestamp) => {
+    const eventIndex = trimmedEvents.findIndex((event) => {
+      return `${event.category}.${event.name}` === eventName && event.timestamp === timestamp;
+    });
+
+    listRefs.current[eventIndex].scrollIntoView({
+      behavior: 'smooth',
+      block: 'start',
+    });
+
+    // Triggers highlighting the event row that was just selected.
+    setScrollingKey(`${eventName}.${timestamp}`);
+
+    // After 1 second, we want to unset the key, which will remove the highlight.
```

Comment:
Just my opinion, but 1 second seemed a bit fast from watching your demo a couple of times. Did you experiment with slightly longer times or is there a mechanical reason for the 1 second you chose?

Reply:
So this comment is actually incorrect. The timing for the animation is actually controlled in the CSS. The `item-highlight` CSS class controls the timing via an animation.

This 1 second just unsets the value in state so that we can set this to a different key later. I am going to update the comment to reflect what is actually happening.